File: src/main/java/sk/hudak/prco/dao/db/GroupOfProductFindEntityDao.java
package sk.hudak.prco.dao.db;

import sk.hudak.prco.model.ProductEntity;

import java.util.List;
import java.util.Optional;

public interface GroupOfProductFindEntityDao {

    List<ProductEntity> findProductsWitchAreNotInAnyGroup();

    long countOfProductsWitchAreNotInAnyGroup();

    long countOfProductInGroup(String groupName);

    Optional<Long> findFirstProductGroupId(Long productId);

    List<Long> findGroupIdsWithProductId(Long productId);
}

File: src/main/java/sk/hudak/prco/dao/db/GroupOfProductFindEntityDao.kt
package sk.hudak.prco.dao.db

import sk.hudak.prco.model.ProductEntity
import java.util.*

interface GroupOfProductFindEntityDao {

    fun findProductsWitchAreNotInAnyGroup(): List<ProductEntity>

    fun countOfProductsWitchAreNotInAnyGroup(): Long

    fun countOfProductInGroup(groupName: String): Long

    fun findFirstProductGroupId(productId: Long?): Optional<Long>

    fun findGroupIdsWithProductId(productId: Long?): List<Long>
}
